src/main/java/nl/nn/testtool/web/api/ApiBase.java

```diff
@@ -49,23 +54,23 @@ protected <T> T getBean(String beanName) {
 	}
 
 	protected void setSessionAttr(String key, Object value) {
-		httpRequest.getSession().setAttribute(key, value);
+		getHttpServletRequest().getSession().setAttribute(key, value);
 	}
 
 	protected <T> T getSessionAttr(String key) {
 		return getSessionAttr(key, true);
 	}
 
 	protected <T> T getSessionAttr(String key, boolean throwException) {
-		T object = (T) httpRequest.getSession().getAttribute(key);
+		T object = (T) getHttpServletRequest().getSession().getAttribute(key);
 		if (object == null && throwException)
 			throw new ApiException("No session attribute with name [" + key + "] found.");
 		return object;
 	}
 
 	@Override
 	public Principal getUserPrincipal() {
-		return httpRequest.getUserPrincipal();
+		return getHttpServletRequest().getUserPrincipal();
```

Comment:
De HttpServerRequest zou je nooit nodig moeten hebben, het ophalen van de gebruiker gaat bijvoorbeeld zo:

```suggestion
		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
		if(authentication != null) {
			return authentication.getName();
		}
		return null;
```